### DISCUSSION TITLE
How to block a contact

### DISCUSSION THREAD

**Discussion by kvvaradha** [netflie/whatsapp-cloud-api#123] [Q&A]
is there any api call to block a contact which tries to spam the account

**Comment by derrickobedgiu1** [comment]
This is irrelevant to the lib but there are tons of methods you can use. For example, a month or two ago I had SPAM accounts that were flooding my paid WhatsApp business plan. In turn, within 24 hours they accumulated over $40 in expenditure on my WhatsApp Account end that Meta had to temporarily ban me for it. Since these were valid numbers that WhatsApp seemed to accept, I decided to ban the entire country from accessing the chatbot due to the frequency at which the new contacts were being registered in the database and the message requests to OpenAi's ChatGPT API....yes I'm on a ChatGPT chatbot within WhatsApp...these were Morocco numbers that were over 2000. I just placed a filter for the country code such that if the incoming request matched Morocco, no response would be sent back to WhatsApp, in turn, Meta won't charge for conversations with a business that has not replied.
Yes, I got charged roughly $30 from OpenAi's end because these contacts were making a lot of requests every after few seconds. You just have to figure out how to filter them, but you're asking for a relevant issue IN THE WRONG PLACE. That's a personal issue, rather than a community issue...

**Comment by kvvaradha** [comment]
I was just asking about the cloud api call function to provide the way to spam or report block a chat.  

I can understand the bot people trying to spam the contacts. its difficult to prevent them. so thats right to block some specific country if you are not having any business to that country. though there are some marketing peoples, they always trying to connect. so i was looking help to report those contacts and mark them as spam or block future messages from that contact 

hope you understand that. so its not community issue, its the programming requirement i am registering here. 

[WhatsApp cloud Reference](https://www.postman.com/meta/workspace/whatsapp-business-platform/request/14291747-329e0e9e-17ae-49f4-b41c-d4b88242d370?ctx=code)
